examples/extensions/update_sitelink_campaign_extension_setting.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Updates the sitelink campaign extension setting to replace its extension feed
+items. Note that this doesn't completely remove your old extension feed items.
+See https://developers.google.com/google-ads/api/docs/extensions/extension-settings/overview
+for details."""
+
+
+import argparse
+import sys
+
+from google.api_core import protobuf_helpers
+import google.api_core.path_template
+
+from google.ads.google_ads.client import GoogleAdsClient
+from google.ads.google_ads.errors import GoogleAdsException
+from google.ads.google_ads.util import ResourceName
+
+
+def main(client, customer_id, campaign_id, feed_item_ids):
+    """The main method that creates all necessary entities for the example.
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        customer_id: a client customer ID.
+        campaign_id: the campaign ID.
+        feed_item_ids: the IDs the feed items to replace.
+    """
+    campaign_extension_setting_service = client.get_service(
+        "CampaignExtensionSettingService", version="v6"
+    )
+    campaign_extension_setting_operation = client.get_type(
+        "CampaignExtensionSettingOperation", version="v6"
+    )
+    extension_feed_item_service = client.get_service(
+        "ExtensionFeedItemService", version="v6"
+    )
+
+    extension_feed_items = map(
+        lambda feed_item_id: extension_feed_item_service.extension_feed_item_path(
+            customer_id, feed_item_id
+        ),
+        feed_item_ids,
+    )
+    campaign_extension_setting = campaign_extension_setting_operation.update
+    extension_type_enum = client.get_type("ExtensionTypeEnum", version="v6")
+    composite_id = ResourceName.format_composite(
+        # the docs don't make it clear that this should reference the name of the extension type and not the enum
```

Comment:
@wihl @BenRKarl I'll need to remove this comment but first wanted to see if this way of using the enum is expected

Reply:
Yep this is correct - it's the best way to access the enum value name, (i.e. "SITELINK") from the Enum itself, without simply hardcoding the string directly. It's a little roundabout but I haven't found a simpler way to do it.